assets/app/scripts/services/metrics.js

```diff
@@ -0,0 +1,122 @@
+      uptime: RATE_TEMPLATE
+    };
+
+    var metricsURL;
+    function getMetricsURL() {
+      if (angular.isDefined(metricsURL)) {
+        return $q.when(metricsURL);
+      }
+
+      return APIDiscovery.getMetricsURL().then(function(url) {
+        // Remove trailing slash if present.
+        metricsURL = (url || '').replace(/\/$/, "");
+        return metricsURL;
+      });
+    }
+
+    function normalizeMetricsData(data, metric) {
+      // Track the previous value for CPU usage calculations.
+      var lastValue;
+      if (!data.length) {
+        return;
+      }
+
+      angular.forEach(data, function(point) {
+        // Calculate a timestamp based on the midtime if missing.
+        if (!point.timestamp) {
+          point.timestamp = point.start + (point.end - point.start) / 2;
+        }
+
+        // Set point.value to the average or 0 if missing.
+        if (!point.value || point.value === "NaN") {
+          point.value = point.avg || 0;
+        }
+
+        // Convert cumulative CPU usage in nanoseconds to millicores.
+        if (metric === 'cpu/usage') {
+          var value = point.value;
+          if (angular.isDefined(lastValue)) {
+            var timeInMillis = point.end - point.start;
+            // Find the usage for just this bucket by comparing it to the last value.
+            // Values are in nanoseconds. Calculate usage in millis.
+            var usageInMillis = (value - lastValue) / 1000000;
+            // Convert to millicores.
+            point.value = (usageInMillis / timeInMillis) * 1000;
+          }
+
+          lastValue = value;
+        }
+      });
+
+      // Remove the first value since it can't be used CPU utilization.
+      // We want the same number of data points for all charts.
+      data.shift();
+      return data;
+    }
+
+    return {
+      // Check if the metrics service is available. The service is considered
+      // available if a metrics URL is set. Returns a promise resolved with a
+      // boolean value.
+      isAvailable: function() {
+        return getMetricsURL().then(function(url) {
+          return !!url;
+        });
+      },
+
+      // Get metrics data for a container.
+      //
+      // Keyword Args
+      //   pod:            the pod object
+      //   containerName:  the container name
+      //   metric:         the metric to check, e.g. "memory/usage"
+      //   start:          start time in millis
+      //   end:            end time in millis
+      //
+      // Returns a promise resolved with the metrics data.
+      get: function(kwArgs) {
+        return getMetricsURL().then(function(metricsURL) {
+          var reqURL,
+              template = metricsURL + templateByMetric[kwArgs.metric];
+
+          reqURL = URI.expand(template, {
+            podUID: kwArgs.pod.metadata.uid,
+            containerName: kwArgs.containerName,
+            metric: kwArgs.metric
+          }).toString();
+
+          Logger.log("Requesting metrics:", reqURL);
```

Comment:
If this is going to generate a log every 30 seconds for each chart then we may want to make a sub logger like we do for auth that can be enabled independently from the main logger.